viamonvif/discovery.go

```diff
@@ -328,7 +358,109 @@ func GetMediaInfoFromProfiles(
 			},
 			Codec: string(profile.VideoEncoderConfiguration.Encoding),
 		})
+
+		// Check if the profile has PTZ capabilities
+		if profile.PTZConfiguration.Token == "" {
+			logger.Debugf("Profile %s does not have PTZ capabilities, %s", profile.Name, streamURI.String())
+			continue
+		}
+
+		logger.Debugf("Profile %s has PTZ capabilities, %s", profile.Name, streamURI.String())
+
+		nodes, err := dev.GetPTZNodes(ctx)
+		if err != nil {
+			logger.Warnf("Unable to determine onvif ptz nodes for profile %s: %s, err: %s",
+				profile.Name, streamURI.String(), err)
+			continue
+		}
+
+		ptzCfg := profile.PTZConfiguration
+		nodeToken := string(ptzCfg.NodeToken)
+
+		// find the matching PTZNode
+		var selected *onvif.PTZNode
+		for i := range nodes {
+			if string(nodes[i].Token) == nodeToken {
+				selected = &nodes[i]
+				break
+			}
+		}
+		if selected == nil {
+			logger.Warnf("PTZ node %s not found for profile %s", nodeToken, profile.Name)
+			continue
+		}
+
+		// Helpers to turn ONVIF spaces into config types
+		make2D := func(d onvif.Space2DDescription) *ptzclient.PanTiltSpace {
+			if d.URI == "" {
```

Comment:
Do we need to check URI is empty here? If so, would it make sense to check all ZoomSpace and PanTiltSpace values?

Reply:
The URI check here is simply to check if the space is defined. I do not think we need to verify the `XRange` and `YRange` values as well.
(If the URI is defined we assume the space is well formatted and all range values are there)
This is working well in manual testing.